Multiqueue fixes

wingo opened the pull request:
THis PR builds on #972 and fixes a few errors.  With this, running the icmp_on_fail.conf on 83:00.{0,1} like this:

```
$ make && sudo ./snabb lwaftr run --name lwaftr --cpu 7-11 --conf program/lwaftr/tests/data/icmp_on_fail.conf --v4 83:00.0 --v6 83:00.1
```

I can then get the config:

```
$ make && sudo ./snabb config get lwaftr /softwire-config/instance
make: 'snabb' is up to date.
{
  device 83:00.0;
  queue {
    id 0;
    external-interface {
      device 83:00.1;
      ip 10.10.10.10;
      mac 12:12:12:12:12:12;
      next-hop {
        mac 68:68:68:68:68:68;
      }
    }
    internal-interface {
      ip 8:9:a:b:c:d:e:f;
      mac 22:22:22:22:22:22;
      next-hop {
        mac 44:44:44:44:44:44;
      }
    }
  }
}
```

And add an RSS worker:

```
$ cat second-queue.conf
{
  id 1;
  external-interface {
    device 83:00.1;
    ip 10.10.10.10;
    mac 12:12:12:12:12:12;
    next-hop {
      mac 68:68:68:68:68:68;
    }
  }
  internal-interface {
    ip 8:9:a:b:c:d:e:f;
    mac 22:22:22:22:22:22;
    next-hop {
      mac 44:44:44:44:44:44;
    }
  }
}

$ make && sudo ./snabb config add lwaftr /softwire-config/instance[device=83:00.0]/queue < second-queue.conf
make: 'snabb' is up to date.
lib/yang/binary.lua:159: attempt to index local 'data' (a nil value)
```

OK so weird error there, need to look into it.  However:

```
$ sudo ./snabb ps
2875
19769
20061
24491
25025	[lwaftr] * [schema: snabb-softwire-v2]
  \- 25031   worker for 25025
  \- 25175   worker for 25025

$ make && sudo ./snabb config get lwaftr /softwire-config/instance
make: 'snabb' is up to date.
{
  device 83:00.0;
  queue {
    id 0;
    external-interface {
      device 83:00.1;
      ip 10.10.10.10;
      mac 12:12:12:12:12:12;
      next-hop {
        mac 68:68:68:68:68:68;
      }
    }
    internal-interface {
      ip 8:9:a:b:c:d:e:f;
      mac 22:22:22:22:22:22;
      next-hop {
        mac 44:44:44:44:44:44;
      }
    }
  }
  queue {
    id 1;
    external-interface {
      device 83:00.1;
      ip 10.10.10.10;
      mac 12:12:12:12:12:12;
      next-hop {
        mac 68:68:68:68:68:68;
      }
    }
    internal-interface {
      ip 8:9:a:b:c:d:e:f;
      mac 22:22:22:22:22:22;
      next-hop {
        mac 44:44:44:44:44:44;
      }
    }
  }
}
```

wingo committed:
Fix bug printing error in config leader

wingo committed:
Fix accessing first queue

This commit adds a common lwutil.parse_instance function and uses it
to access the first queue.  The first queue is not guaranteed to be at
position 1 in the values array; position 1 could have been filled but
later deleted.

takikawa commented:
I think the error in `binary.lua` might be significant, because I tried to run this with a modified `icmp_on_fail.conf` with the extra queue and in that case it fails to bring up the lwaftr. Seems to get added with `config add` but maybe the extra queue isn't running correctly in that case.

(also trying to run this led me to find some bugs in the `intel_mp` part, which I'm fixing now. BTW, I also think the multiqueue code should set `poolnum` for the driver---otherwise new queues would get assigned to new VMDq pools instead of attaching to an existing pool)

wingo commented:
Sorry for the inconvenience but I rebased this one due to rebase in #972.  Previous build failure log here: https://gist.github.com/lwaftr-igalia/eff75040758a2296c47470ed8a240ca4

wingo committed:
Fix sparse cltable value serialization

It's possible that cltable values aren't packed, so we need to serialize
value indices in addition to values.  Bump binary version as well.

wingo committed:
Fix insertion into sparse cltables

Previously we would do a table.insert, then assume the index was
inserted at #values.  However the # operator just returns a filled
index for which the next index is nil, and thus table.insert could
insert into the middle of a table but we don't know afterwards where
that insertion was!  So, instead use # to find an insertion point, and
don't use table.insert.

wingo committed:
Fix serialization of sparse arrays, again

wingo committed:
Fix yet another sparse cltable serialization problem (!)
